### ISSUE TITLE
Unable to retrieve Decimal128 when using bson-ext

### ISSUE THREAD

**Issue by azhang66** [Automattic/mongoose#5850]
**Node:** 9.2.0
**Mongoose:** 4.13.4
**MongoDB:** 3.4.10
**Platform:** Linux 4.13.12 x86_64

Mongoose is unable to retrieve Decimal128 attributes when `bson-ext` is installed. Insertion works fine, but retrieval does not. Consider the following code:
```javascript
mongoose = require('mongoose');

mongoose.Promise = Promise;
var db = mongoose.connect('mongodb://localhost:27017/test', { useMongoClient: true });

var Doc = mongoose.model('Test', new mongoose.Schema({
    x: mongoose.Schema.Types.Decimal128
}));

Doc.create({ x: '0.1' }).
  then(doc => doc.update({ $inc: { x: '0.2' } }).then(() => doc)).
  then(doc => Doc.findById(doc)).
  then(doc => console.log('doc', doc.toObject())).
  then(() => { db.close(); }).
  catch(error => console.error(error));
```
**The expected output (when `bson-ext` is not installed)**
```
doc { _id: 5a177431453e4c7ed0bc5644,
  x: { '$numberDecimal': '0.3' },
  __v: 0 }
```
**However, after installing `bson-ext`...**
```
$ npm install --save bson-ext
```
**the following output occurs:**
```
doc { _id: 5a1774dc3aa78d0144c62bc3, __v: 0 }
```

**Expected behavior:** Mongoose returns the same data both times
**Erroneous behavior:** Mongoose fails to return the Decimal128 attribute when `bson-ext` is installed.

**Comment by vkarpov15** [comment]
Thanks for the bug report, will investigate ASAP

**Comment by vkarpov15** [comment]
Thanks for the detailed repro, we found the issue and merged a fix into master. Will be released with 4.13.7 later this week :+1:
